src/libstd/ffi/c_str.rs

```diff
@@ -266,11 +266,17 @@ impl CString {
 
     /// Transfers ownership of the string to a C caller.
     ///
-    /// The pointer must be returned to Rust and reconstituted using
-    /// `from_raw` to be properly deallocated. Specifically, one
+    /// Unless the same memory allocator is used for both Rust and C,
+    /// the acquired pointer must be returned to Rust and reconstituted
+    /// using `from_raw` to be properly deallocated. Specifically, one
```

Comment:
Doesn't this also imply a corresponding change to the `from_raw` contract as well? Specifically this part:

> Other usage (e.g. trying to take ownership of a string that was allocated by foreign code) is likely to lead to undefined behavior or allocator corruption.